paddle/phi/kernels/multi_tensor_adam_utility_kernel.cuh

```diff
@@ -0,0 +1,139 @@
+#include "paddle/phi/backends/gpu/gpu_context.h"
+#include "paddle/phi/core/kernel_registry.h"
+#include "paddle/fluid/framework/tensor_util.h"
+#include "paddle/phi/core/tensor_utils.h"
+
+#include <assert.h>
+
+namespace phi {
+
+#define MAX_COMPUTE_GROUP_SIZE 65535
+
+constexpr int max_tensors_size[6] = {110, 64, 48, 36, 30, 24};
+constexpr int max_blocks_size[6] = {320, 320, 320, 320, 320, 320};
+
+template<int n> struct TensorAndBlockInf
+{
+  void* tensors_addr[n-1][max_tensors_size[n-1]];
+  const void* grad[max_tensors_size[n-1]];
+  int sizes[max_tensors_size[n-1]];
+  unsigned char tenosr_for_this_block[max_blocks_size[n-1]];
+  //int16
+  unsigned short int compute_group_for_this_block[max_blocks_size[n-1]];
+  int start_compute_group_this_tensor;
+};
+
+
+template<typename MT, typename T, typename U, typename... ArgTypes>
+__global__ void multi_tensor_adam_utility_kernel(
+    int compute_group_size,
+    T tabi,
+    U cuda_kernel_functor_op,
+    ArgTypes... args)
+{
+  cuda_kernel_functor_op(compute_group_size, tabi, args...);
+}
+
+template<int input_num, typename MT, typename T, typename Context, typename... ArgTypes>
+void  multi_tensor_adam_utility(
+  const Context &dev_ctx,
+  int block_size, //512
+  int compute_group_size, //2048*32
+  const std::vector<std::vector<DenseTensor *>> & tensor_and_block_inf,
+  const std::vector<const DenseTensor *> &g,
+  T cuda_kernel_functor_op,
+  ArgTypes... args)
+{
+  PADDLE_ENFORCE_EQ(
+         tensor_and_block_inf.size(),
+        input_num - 1,
+        errors::InvalidArgument("ensor_lists.size() != input_num - 1"));
```

Comment:
The error messages should be polished (polish the following other error message as well). Supposing you are an end-user and you see the error message above, you do not know what the error message means and how to fix it.

BTW, `ensor_lists.size()` -> `tensor_lists.size()`